tests/multigrid-global-coarsening/multigrid_util.h

```diff
@@ -269,6 +280,8 @@ class Operator : public Subscriptor
   AffineConstraints<number> constraints;
 
   mutable TrilinosWrappers::SparseMatrix system_matrix;
+
+  std::vector<unsigned int> constrained_indices;
```

Comment:
It is only a test, so this is fine. Otherwise, I would recommend to set a bool flag to document more clearly and use `matrix_free.get_constrained_dofs()` in the `vmult` function according to this flag, to make the code more self-explanatory.